pkg/uninstall/uninstall.go

```diff
@@ -52,6 +54,32 @@ func (obj *uninstaller) Uninstall(ctx context.Context, pkg ctrlpkg.Package, isDr
 	return obj.delete(ctx, pkg, isDryRun)
 }
 
+func (obj *uninstaller) IsNamespaceSafeToDelete(ctx context.Context, pkg ctrlpkg.Package) error {
+	var packages v1alpha1.PackageList
+	err := obj.client.Packages(pkg.GetNamespace()).GetAll(ctx, &packages)
+	if err != nil {
+		return err
+	}
+	if len(packages.Items) > 1 {
+		return fmt.Errorf("namespace %s contains more than one package", pkg.GetNamespace())
+	}
+	return nil
+}
+
+func (obj *uninstaller) UninstallAndDeleteNamespaceBlocking(
+	ctx context.Context,
+	pkg ctrlpkg.Package,
+	isDryRun bool,
+) error {
+	namespace := pkg.GetNamespace()
+	if err := obj.UninstallBlocking(ctx, pkg, isDryRun); err != nil {
+		return err
+	}
+	clientset := clicontext.KubernetesClientFromContext(ctx)
+
+	return obj.deleteNamespaceBlocking(ctx, clientset, namespace)
+}
```

Comment:
I think it would be simpler to add a "DeleteNamespace" option to the existing UninstallBlocking function